sparse/coo/umath.py

```diff
@@ -3,8 +3,9 @@
 import numpy as np
 import scipy.sparse
 
-from ..utils import isscalar, PositinalArgumentPartial, _zero_of_dtype
+from .src.umath import pywhere
```

Comment:
Why define and import this in another module?  Can we define it here instead?  It would be nice to reduce the number of indirections that someone new to this codebase would have to make in order to find functionality.

Reply:
Ah, it would be especially nice to remove the `src` submodule.  While these are quite common in other languages I rarely find them in Python code in the pydata ecosystem.  In general the community seems to have preferred a somewhat flat hierarchy when possible.

Reply:
I've separated it for these reasons:

- coverage.py, CodeCov et.al. can't produce coverage for Numba JIT-ed functions. Keeping it separate lets us ignore it in coverage.
- Keeping Numba code separate (for a number of reasons I can't coherently put down in words)

Reply:
I would prefer to keep things flat if possible.  I think that the submodule and file hierarchy are more important than coverage metrics, so coverage considerations shouldn't determine our hierarchy here.  You might consider using a [`#pragma: no cover`](http://coverage.readthedocs.io/en/coverage-4.3.4/excluding.html) comment instead.

I think that one of the advantages of Numba is that we *can* include it in our normal code and don't need to treat it as something special.  Removing this separation between low-level efficient code and normal code is, I think, important.

Reply:
... I think that keeping these things together is especially important for new developers or users of this code as they try to navigate things.   I agree that this isn't much of an issue for us as we're writing things, I do think that splitting things up and requiring navigation of a codebase affects readability for others though.

Reply:
Sounds like a good idea. My main concern was code coverage, the pragma seems like a good way to handle that.